tests/test_transaction.py

```diff
@@ -8,20 +8,25 @@
 )
 from bitcash.utils import hex_to_bytes
 from bitcash.wallet import PrivateKey
-from .samples import WALLET_FORMAT_MAIN, BITCOIN_CASHADDRESS_TEST_COMPRESSED
+from .samples import (
+    WALLET_FORMAT_MAIN, BITCOIN_CASHADDRESS, BITCOIN_CASHADDRESS_COMPRESSED
+)
 
 
-RETURN_ADDRESS = 'n2eMqTT929pb1RDNuqEnxdaLau1rxy3efi'
+RETURN_ADDRESS = BITCOIN_CASHADDRESS
 
 FINAL_TX_1 = ('01000000018878399d83ec25c627cfbf753ff9ca3602373eac437ab2676154a'
-              '3c2da23adf3010000008a47304402204d6f28d77fa31cfc6c13bb1bda2628f2'
-              '237e2630e892dc62bb319eb75dc7f9310220741f4df7d9460daa844389eb23f'
-              'b318dd674967144eb89477608b10e03c175034141043d5c2875c9bd116875a7'
+              '3c2da23adf3010000008a473044022013d0751fb16eb7e1ac75aa799ebbfb55'
+              '2bf112174f4a7eea689b8930cf582abc02206bc2082f38019476ecc3dc839fd'
+              'eda87a70041f23f0843edd101792109a10c6e4141043d5c2875c9bd116875a7'
               '1a5db64cffcb13396b163d039b1d932782489180433476a4352a2add00ebb0d'
               '5c94c515b72eb10f1fd8f3f03b42f4a2b255bfc9aa9e3ffffffff0250c30000'
-              '000000001976a914e7c1345fc8f87c68170b3aa798a956c2fe6a9eff88ac088'
-              '8fc04000000001976a91492461bde6283b461ece7ddf4dbf1e0a48bd113d888'
+              '000000001976a91492461bde6283b461ece7ddf4dbf1e0a48bd113d888ac088'
+              '8fc04000000001976a914990ef60d63b5b5964a1c2282061af45123e93fcb88'
               'ac00000000')
+
+FINAL_TX_ID = "f7e1086edae26fee6ad5efcf9d2c2ff6126c468ddc3a35ed4df51ad557b56ce0"
```

Comment:
This is different than what we currently have. Why is that?

Reply:
This "test_transaction" was set up very strangely. It was using legacy format **testnet** addresses, but importing the **mainnet** WIF and constructing transactions that way. See:

<img width="1058" alt="legacy" src="https://user-images.githubusercontent.com/22279374/97528196-b4ffda80-19df-11eb-9cb1-e083fcedbd28.png">

To be honest that almost seems like a bug, since no enforcement is happening to prevent this mainnet/testnet crossover. But since I removed legacy address support, this test was obviously failing. I updated the addresses to cash addresses instead, as well as switching everything to be purely 'mainnet' for consistency. This switch to strictly mainnet changed the hashes in the tx outputs, which in turn changed the transaction ID.

Additionally I replaced several hard-coded uses of addresses with the proper variables from the "samples" file. One hard-coded legacy address it used was `n2eMqTT929pb1RDNuqEnxdaLau1rxy3efi` which I don't see referenced anywhere else (including in the samples). Since the hashes were different now anyway, I adjusted this test to use BITCOIN_CASHADDRESS and BITCOIN_CASHADDRESS_COMPRESSED instead, since these two different addresses *are* included in the samples.

Most of the functions used for constructing transactions are the same between mainnet and testnet, but I'm curious what your thoughts are on adding separate test for both (and using their respective addresses)? I could do that in a separate 'maintenance' PR if desired. (it's not really in the scope of this one.)

Reply:
Ah, thank you for catching that.

I think though that Bitcoin Cash only allows compressed addresses. That's maybe another thing we can remove. I think uncompressed address transactions are dropped by the network, but maybe I'm mistaken.

Reply:
I would need to dig a bit more into it, but I'm pretty certain both compressed and uncompressed can actually be used. When I was working with regtest addresses and editing the unit tests, there are multiple places were both types of addresses are used and work as expected.

Reply:
Interesting, ok. Not sure why anyone would want to use uncompressed except to get coins off those addresses that hadn't moved since the fork.